jena-arq/src/main/java/org/apache/jena/sparql/syntax/syntaxtransform/ExprTransformApplyElementTransform.java

```diff
@@ -43,7 +44,15 @@ public ExprTransformApplyElementTransform(ElementTransform transform, boolean al
     @Override
     public Expr transform(ExprFunctionOp funcOp, ExprList args, Op opArg)
     {
-        Element el2 = ElementTransformer.transform(funcOp.getElement(), transform);
+        // If the element is null then an attempt is made to obtain it from the algebra.
+        // A given element takes precedence over the algebra.
+        Element el1 = funcOp.getElement();
+        if (el1 == null) {
+            Op op = funcOp.getGraphPattern();
+            el1 = op == null ? null : OpAsQuery.asElement(op);
+        }
+        // ElementTransformer will warn should it happen that null is passed to it.
+        Element el2 = ElementTransformer.transform(el1, transform, this);
```

Comment:
This is now the updated actual fix - together with the introduction of `OpAsQuery.asElement`.